hibernate-envers/src/main/java/org/hibernate/envers/internal/entities/mapper/id/AbstractIdMapper.java

```diff
@@ -67,13 +67,16 @@ public void addIdsEqualToQuery(Parameters parameters, String prefix1, IdMapper m
 			final QueryParameterData paramData1 = paramDataIter1.next();
 			final QueryParameterData paramData2 = paramDataIter2.next();
 
-			parametersToUse.addWhere(
-					paramData1.getProperty( prefix1 ),
-					false,
-					"=",
-					paramData2.getProperty( prefix2 ),
-					false
-			);
+			String p1Prop = paramData1.getProperty( prefix1 );
+			String p2Prop = paramData2.getProperty( prefix2 );
+
+			// (p1.prop = p2.prop or (p1.prop is null and p2.prop is null))
+			Parameters sub1 = parametersToUse.addSubParameters( "or" );
+			sub1.addWhere( p1Prop, false, "=", p2Prop, false );
+			Parameters sub2 = sub1.addSubParameters( "and" );
+			sub2.addNullRestriction( p1Prop, false );
+			sub2.addNullRestriction( p2Prop, false );
```

Comment:
I'd probably recommend we unify this logic into the query `Parameters` class so that basically if we ever need to reuse the notion of `WHERE ( x=y OR x is NULL and y IS NULL )` we have that ability.

So I would rather add a method to `org.hibernate.envers.internal.tools.query.Parameters` as follows:
<pre>
public void addWhereOrNullRestriction(String left, boolean addAliasLeft, String op, String right, boolean addAliasRight) {
  // apply the normal addWhere predicate
  final Parameters sub1 = addSubParameters( "or" );
  sub1.addWhere( left, addAliasLeft, op, right, addAliasRight );
  // apply the is null predicate for both join properties
  final Parameters sub2 = sub1.addSubParameters( "and" );
  sub2.addNullRestriction( left, false );
  sub2.addNullRestriction( right, false );
}
</pre>

Then `AbstractIdMapper` can simply call `addWhereOrNullRestriction` rather than just `addWhere`.